cmd/sidecars/network-passt-binding/server/server.go

```diff
@@ -0,0 +1,111 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"kubevirt.io/kubevirt/cmd/sidecars/network-passt-binding/callback"
+	"kubevirt.io/kubevirt/cmd/sidecars/network-passt-binding/domain"
+
+	vmschema "kubevirt.io/api/core/v1"
+
+	hooksInfo "kubevirt.io/kubevirt/pkg/hooks/info"
+	hooksV1alpha2 "kubevirt.io/kubevirt/pkg/hooks/v1alpha2"
+
+	"kubevirt.io/client-go/log"
+)
+
+type InfoServer struct {
+	Version string
+}
+
+func (s InfoServer) Info(_ context.Context, _ *hooksInfo.InfoParams) (*hooksInfo.InfoResult, error) {
+	infoResult := &hooksInfo.InfoResult{
+		Name: "network-passt-binding",
+		Versions: []string{
+			s.Version,
+		},
+		HookPoints: []*hooksInfo.HookPoint{
+			{
+				Name:     hooksInfo.OnDefineDomainHookPointName,
+				Priority: 0,
+			},
+		},
+	}
+
+	log.Log.Infof("hook info result: %+v", infoResult)
+
+	return infoResult, nil
+}
+
+type V1alpha2Server struct {
+	LogFile    string
+	QueueCount uint
+}
+
+func (s V1alpha2Server) OnDefineDomain(_ context.Context, params *hooksV1alpha2.OnDefineDomainParams) (*hooksV1alpha2.OnDefineDomainResult, error) {
+	vmi := &vmschema.VirtualMachineInstance{}
+	if err := json.Unmarshal(params.GetVmi(), vmi); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal VMI: %v", err)
+	}
+
+	useVirtioTransitional := vmi.Spec.Domain.Devices.UseVirtioTransitional != nil && *vmi.Spec.Domain.Devices.UseVirtioTransitional
+	useLaunchSecurity := vmi.Spec.Domain.LaunchSecurity != nil && vmi.Spec.Domain.LaunchSecurity.SEV != nil
+
+	const istioInjectAnnotation = "sidecar.istio.io/inject"
+	istioProxyInjectionEnabled := false
+	if val, ok := vmi.GetAnnotations()[istioInjectAnnotation]; ok {
+		istioProxyInjectionEnabled = strings.ToLower(val) == "true"
+	}
+
+	opts := domain.NetworkConfiguratorOptions{
+		UseVirtioTransitional:      useVirtioTransitional,
+		UseLaunchSecurity:          useLaunchSecurity,
+		IstioProxyInjectionEnabled: istioProxyInjectionEnabled,
+		LogFile:                    s.LogFile,
+		QueueCount:                 s.QueueCount,
+	}
+
+	passtConfigurator, err := domain.NewPasstNetworkConfigurator(vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create passt configurator: %v", err)
+	}
+
+	newDomainXML, err := callback.OnDefineDomain(params.GetDomainXML(), passtConfigurator)
+	if err != nil {
+		return nil, err
+	}
+
+	return &hooksV1alpha2.OnDefineDomainResult{
+		DomainXML: newDomainXML,
+	}, nil
+}
+
+func (s V1alpha2Server) PreCloudInitIso(_ context.Context, params *hooksV1alpha2.PreCloudInitIsoParams) (*hooksV1alpha2.PreCloudInitIsoResult, error) {
+	log.Log.Infof("PreCloudInitIso hook is not implemented, returning the given cloudInitData")
```

Comment:
I still think this one should be dropped, it has no value.

Reply:
Done